app/src/main/java/org/coepi/android/ui/symptoms/breathless/BreathlessViewModel.kt

```diff
@@ -0,0 +1,90 @@
+import org.coepi.android.R.string.symptom_report_breathlessness_cause_option_walking_yards_or_mins_on_ground
+import org.coepi.android.common.Optional
+import org.coepi.android.common.Optional.None
+import org.coepi.android.common.toNullable
+import org.coepi.android.domain.symptomflow.SymptomFlowManager
+import org.coepi.android.domain.symptomflow.SymptomInputs.Breathlessness
+import org.coepi.android.domain.symptomflow.SymptomInputs.Breathlessness.Cause.EXERCISE
+import org.coepi.android.domain.symptomflow.SymptomInputs.Breathlessness.Cause.GROUND_OWN_PACE
+import org.coepi.android.domain.symptomflow.SymptomInputs.Breathlessness.Cause.HURRY_OR_HILL
+import org.coepi.android.domain.symptomflow.SymptomInputs.Breathlessness.Cause.LEAVING_HOUSE_OR_DRESSING
+import org.coepi.android.domain.symptomflow.SymptomInputs.Breathlessness.Cause.WALKING_YARDS_OR_MINS_ON_GROUND
+import org.coepi.android.extensions.rx.toLiveData
+import org.coepi.android.system.Resources
+import org.coepi.android.ui.navigation.NavigationCommand.Back
+import org.coepi.android.ui.navigation.RootNavigation
+
+class BreathlessViewModel(
+    private val navigation: RootNavigation,
+    private val resources: Resources,
+    private val symptomFlowManager: SymptomFlowManager
+) : ViewModel() {
+
+    private val selectedCause: BehaviorSubject<Optional<Breathlessness.Cause>> = createDefault(None)
+
+    private val coughStatuses : List<Breathlessness.Cause> = listOf(
+        LEAVING_HOUSE_OR_DRESSING, WALKING_YARDS_OR_MINS_ON_GROUND, GROUND_OWN_PACE, HURRY_OR_HILL, EXERCISE
+    )
+
+    private val causesObservable: Observable<List<BreathlessViewData>> = selectedCause
+        .map { selectedCause ->
+            coughStatuses.map { it.toViewData(isChecked = it == selectedCause.toNullable()) }
+        }
+
+    val causes: LiveData<List<BreathlessViewData>> = causesObservable.toLiveData()
+
+    fun onClickSkip() {
+        symptomFlowManager.navigateForward()
+    }
+
+    fun onBack() {
+        symptomFlowManager.onBack()
+    }
+
+    fun onBackPressed() {
+        onBack()
+        navigation.navigate(Back)
+    }
+
+    private fun Breathlessness.Cause.toViewData(isChecked: Boolean): BreathlessViewData =
+        BreathlessViewData(name(), icon(), isChecked, this)
+
+    private fun Breathlessness.Cause.name(): String = when (this) {
+        LEAVING_HOUSE_OR_DRESSING ->  symptom_report_breathlessness_cause_option_leaving_house_or_dressing
+        WALKING_YARDS_OR_MINS_ON_GROUND ->  symptom_report_breathlessness_cause_option_walking_yards_or_mins_on_ground
+        GROUND_OWN_PACE ->  symptom_report_breathlessness_cause_option_ground_own_pace
+        HURRY_OR_HILL ->  symptom_report_breathlessness_cause_option_hurry_or_hill
+        EXERCISE ->  symptom_report_breathlessness_cause_option_exercise
+    }.let { resources.getString(it) }
+
+    private fun Breathlessness.Cause.icon(): Drawable = when (this) {
+        LEAVING_HOUSE_OR_DRESSING ->  ic_breathless_house
```

Comment:
Let's try to solve this without including the line and circle in the images. This makes it difficult to add new items, reorder or restyle.

In the XML you could add:
- View for "top segment"
- View for "bottom segment"
- View for circle (this can be probably the background of the image view)

The view data would contain data representing the configuration of these views, something like:

data class IconConfig(topSegment: Color, bottomSegment: Color, circleColor: Color, drawable: Drawable?)

If any of the segments is invisible, you could set the color to transparent. Or make the field nullable.

If it's not easy to change the color of circle border, you could create a new drawable per color.

Reply:
Alrighty, removed the line and circles from the breathless icons. The circle is now a background for the icons and the bar is a background to the recycler.

https://github.com/Co-Epi/app-android/pull/152/commits/5c705785e0cd0162f36b04afeee28e46ad28e480